kernel/byterun/coq_uint63_native.h

```diff
@@ -64,17 +64,16 @@
   else \
     accu = uint63_zero(); \
   }while(0)
-/* FIXME this definition is the same as lsr for now */
 #define Uint63_asr(x,y) do{ \
     value uint63_asr_y__ = (y); \
-    if (uint63_asr_y__ < (uint64_t) 127) \
-      accu = (value)(((uint64_t)(x) >> uint63_of_value(uint63_asr_y__)) | 1); \
+    if (uint63_asr_y__ < (int64_t) 127) \
```

Comment:
This is wrong. By comparing `y` to an unsigned bound, this test ensures that `y` is between 0 and 62. If you switch to a signed bound, `y` is still guaranteed to be smaller than 63, but it can now be negative, which leads to undefined behavior later.

By the way, this also means that the `asr.v` testsuite file should contain some shift by negative values, to make sure that 0 is properly returned in that case.

Reply:
Nice catch, thanks. It should be safe to change that line back to `if (uint63_asr_y__ < (uint64_t) 127) \` then, I guess.

I will add such a test.